qiskit_nature/second_q/circuit/library/ansatzes/succd.py

```diff
@@ -146,27 +166,40 @@ def generate_excitations(
         )
         logger.debug("Generated list of single alpha excitations: %s", alpha_excitations)
 
-        # Find all possible double excitations constructed from the list of single excitations.
-        # Note, that we use `combinations_with_replacement` here, in order to also get those double
-        # excitations which excite from the same occupied level twice. We will need those in the
-        # following post-processing step.
-        pool = itertools.combinations_with_replacement(alpha_excitations, 2)
-
-        for exc in pool:
-            # find the two excitations (Note: SUCCD only works for double excitations!)
-            alpha_exc, second_exc = exc[0], exc[1]
-            # shift the second excitation into the beta-spin orbital index range
-            beta_exc = (
-                second_exc[0] + beta_index_shift,
-                second_exc[1] + beta_index_shift,
+        if self._mirror:  # if include the symmetrically mirrored double excitations
```

Comment:
That is not really what I meant. I rather meant a similarly extensive comment as inside the `else` statement which explains why in this case, we can use the "standard" implementation rather than needing to use `combinations_with_replacement` as explained there